tests/resource/test_error/test_error.py

```diff
@@ -181,3 +186,98 @@ class NoErrorOp(ErrorOperation):
                 num_wires = 3
 
             _ = NoErrorOp(wires=[1, 2, 3])
+
+
+class MultiplicativeError(AlgorithmicError):
+    """Multiplicative error object"""
+
+    def combine(self, other):
+        return self.__class__(self.error * other.error)
+
+    def __repr__(self):
+        """Return formal string representation."""
+        return f"MultiplicativeError({self.error})"
+
+
+class AdditiveError(AlgorithmicError):
+    """Additive error object"""
+
+    def combine(self, other):
+        return self.__class__(self.error + other.error)
+
+    def __repr__(self):
+        """Return formal string representation."""
+        return f"AdditiveError({self.error})"
+
+
+class CustomErrorOp1(ErrorOperation):
+    """Custome error operation with multiplicative error"""
+
+    def __init__(self, phase, wires):
+        self.phase = phase
+        super().__init__(phase, wires=wires)
+
+    def error(self, *args, **kwargs):
+        return MultiplicativeError(self.phase)
+
+
+class CustomErrorOp2(ErrorOperation):
+    """Custome error with additive error"""
+
+    def __init__(self, flips, wires):
+        self.flips = flips
+        super().__init__(flips, wires=wires)
+
+    def error(self, *args, **kwargs):
+        return AdditiveError(self.flips)
+
+
+_HAMILTONIAN = qml.dot([1.0, -0.5], [qml.X(0) @ qml.Y(1), qml.Y(0) @ qml.Y(1)])
+
+
+class TestSpecAndTracker:
+    """Test capture of ErrorOperation in specs and tracker."""
+
+    # little hack for stopping device-level decomposition for custom ops
+    @staticmethod
+    def preprocess(execution_config=qml.devices.DefaultExecutionConfig):
+        """A vanilla preprocesser"""
+        return qml.transforms.core.TransformProgram(), execution_config
+
+    dev = qml.device("null.qubit", wires=2)
+    dev.preprocess = preprocess.__func__
```

Comment:
Why this is needed?

Reply:
Here `preprocess` is a static method of the test class, so to assign the `func` instead I have to access it like it. I mainly do this to prevent `null.qubit` from using the default preprocess which tries to find decomposition for the custom ops I create.

Reply:
The users need to do this _hack_ as well when they have a similar workflow with custom ops? What goes wrong if dev.preprocess is not updated, i.e., what happens if you remove this _hack_?

Reply:
They need not do any of it as this is just for executing custom ops on null.qubit. Added a TODO to remove it in the future.